### PR TITLE
Adds CITATION.cff && validation action

### PR DESCRIPTION
**Pull request by JKatzwinkel**
maybe we should leave this unmerged until release, as a reminder that we should not forget to add the `date-released` field with the release date.

### CODE REVIEW

**Comment by funkyfuture** [comment]
i'm postpoing this to 0.4. because:

- the repository's URL will change
- we'll setup a GH workflow for releases (#35) that will generate this file
    - let's use this PR for it
    - i've seen a reasonable jinja action around

**Comment by funkyfuture** [comment]
two+ thoughts:

- the rendered file should be included in the Python package
- instead of including this in the release workflow directly, it could be a seperate workflow that will than be included
  - though i don't know yet where else it would be used
  - i'm not sure whether the validation step should be part of the `quality-checks` workflow

**Comment by JKatzwinkel** [comment]
should we render the citation file in the `release` make target then?

**Comment by funkyfuture** [comment]
now after some time working on an automated release workflow, i'm sceptical the effort would be worth it here. also, there's the problem that the tag should be created after the file has been generated. perspectively there will also be the generated xpath parser module. this one would have to be generated before running tests on every platform.

henceforth we should rely on scripts for these jobs. i propose that you move the generation logic to such a script, add a git hook that calls it and move the verification to the quality checks workflow (i wouldn't do this locally).

is there a more or less canonical name for paths with such scripts that is descriptive?

**Comment by JKatzwinkel** [comment]
I usually put stuff like that into `build/` or `bin/`.

**Comment by funkyfuture** [comment]
i was already into setting up something with `pre-commit`, but realised that the two generation processes are quiet different.

> should we render the citation file in the `release` make target then?

is the question that answers my question *when* is the proper time to set the file's contens. 

https://pypi.org/project/jinja-cli/ looks feasible for the poetry dev environment. also, the validation therefore needs to happen locally. https://github.com/citation-file-format/cff-converter-python has a `--validate` flag that acts as subcommand.

File: .github/templates/CITATION.cff.j2 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+cff-version: 1.2.0
+type: software
+message: >
+  If you cite this software, please use these metadata.
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:4]: the specs declare this field as requires, but also specifies a default. what does that mean?

why don't they propose to put that as a comment at the top when it's intended for human consumers?

File: .github/templates/CITATION.cff.j2 [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+    email: funkyfuture@riseup.net
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:9]:
```suggestion
  - name: The delb project posse
```

File: .github/templates/CITATION.cff.j2 [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+license: AGPL-3.0
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:10]:
```suggestion
license: AGPL-3.0
license-url: https://raw.githubusercontent.com/funkyfuture/delb/{{ release_tag }}/LICENSE.txt
```

File: .github/templates/CITATION.cff.j2 [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+version: "{{ release_tag }}"
+repository-code: "https://github.com/funkyfuture/delb"
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:12]:
```suggestion
repository-code: https://github.com/funkyfuture/delb
url: delb.readthedocs.io/
```

File: .github/templates/CITATION.cff.j2 (hunk 2 above)
Comment (funkyfuture) [reply .github/templates/CITATION.cff.j2:9]: they have two keywords for the same thing.

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (JKatzwinkel) [reply .github/templates/CITATION.cff.j2:4]: idk, they probably want us to specify a msg we are happy with to be extracted/rendered by potential integrations with other services. whats the default?

File: .github/templates/CITATION.cff.j2 (hunk 2 above)
Comment (JKatzwinkel) [reply .github/templates/CITATION.cff.j2:9]: i know, but given/family name lets us make explicit its a person

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (funkyfuture) [reply .github/templates/CITATION.cff.j2:4]: https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#message

File: .github/templates/CITATION.cff.j2 (hunk 2 above)
Comment (funkyfuture) [reply .github/templates/CITATION.cff.j2:9]: i meant that there are the keywords `website` and `url` for the same thing in deifferent scopes.

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (JKatzwinkel) [reply .github/templates/CITATION.cff.j2:4]: ah yes. tbh I find the condition "if you use this software then cite" a little bit bossy and therefore chose a different message.

File: .github/templates/CITATION.cff.j2 [hunk 5]
```diff
@@ -0,0 +1,14 @@
+cff-version: 1.2.0
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:1]:
```suggestion
# If you cite this software, please use these metadata.

cff-version: 1.2.0
```

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:4]:
```suggestion
    Aller politischer Unterricht endlich sollte zentriert sein darin,
    daß Auschwitz nicht sich wiederhole.
```

File: .github/templates/CITATION.cff.j2 [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+    email: funkyfuture@riseup.net
+license: AGPL-3.0
+version: "{{ release_tag }}"
+repository-code: "https://github.com/funkyfuture/delb"
+date-released: "{{ release_date }}"
+#  vim: set ts=2 sw=2 tw=500 et ft=yaml :
```
Comment (funkyfuture) [inline .github/templates/CITATION.cff.j2:14]:
```suggestion

...
```

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (funkyfuture) [reply .github/templates/CITATION.cff.j2:4]: ich bin auch überrascht, daß das "Denken und Handeln" im text gar nicht vorkommt. war mir beim lesen gar nicht aufgefallen.

**Comment by funkyfuture** [comment]
i'd move the `templates` folder to the root.

File: pyproject.toml [hunk 7]
```diff
@@ -37,12 +37,13 @@ cssselect = "*"
 lxml = "*"
 requests = {version = "^2.21", optional = true}
 setuptools = "*"
-cffconvert = "^2.0.0"
 
 [tool.poetry.dev-dependencies]
 autoclasstoc = "^1.1.1"
 black = {version = "*", allow-prereleases = true }
+cffconvert = "^2.0.0"
```
Comment (funkyfuture) [inline pyproject.toml:44]: reichlich weit für'n so junges projekt.

**Comment by funkyfuture** [comment]
i have this branch rebased onto main and added another render approach with `just`. i'm afraid force-pushing it will make previous comments less accessible. shall i wait until you settled the contents?

File: .github/templates/CITATION.cff.j2 (hunk 1 above)
Comment (funkyfuture) [reply .github/templates/CITATION.cff.j2:4]: der letzte vorschlag weist auf die zeitgemäßheit 35 jahre alter, trotzkistischer plattentitel (Redskins) hin.

Comment (JKatzwinkel) [reply .github/templates/CITATION.cff.j2]: :fire: :globe_with_meridians:

File: CITATION.cff [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+cff-version: 1.2.0
+type: software
+message: 🔥🌍🔥
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+license: AGPL-3.0
+version: 0.3.0
+repository-code: https://github.com/funkyfuture/delb
```
Comment (JKatzwinkel) [inline CITATION.cff]:
```suggestion
repository-code: https://github.com/delb-xml/delb-py
```

File: CITATION.cff.tmpl [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+cff-version: 1.2.0
+type: software
+message: 🔥🌍🔥
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+license: AGPL-3.0
+version: _VERSION_
+repository-code: "https://github.com/funkyfuture/delb"
```
Comment (JKatzwinkel) [inline CITATION.cff.tmpl]:
```suggestion
repository-code: "https://github.com/delb-xml/delb-py"
```

File: Justfile [hunk 10]
```diff
@@ -34,16 +36,25 @@ pytest:
 	python -m pytest --cov-config .coveragerc --cov=_delb --cov=delb tests
 
 # release the current version on github & the PyPI
-release: tests
-	git tag -f {{version}}
-	git push origin main
-	git push -f origin {{version}}
-	poetry publish --build
+release:
```
Comment (JKatzwinkel) [inline Justfile:39]:
```suggestion
release: tests
```

File: CITATION.cff [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+cff-version: 1.2.0
+type: software
+message: 🔥🌍🔥
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+license: AGPL-3.0
+version: 0.3.0
+repository-code: https://github.com/delb-py/delb
```
Comment (JKatzwinkel) [inline CITATION.cff:12]:
```suggestion
repository-code: https://github.com/delb-xml/delb-py
```

File: CITATION.cff.tmpl [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+cff-version: 1.2.0
+type: software
+message: 🔥🌍🔥
+title: delb
+authors:
+  - family-names: Sachsenheim
+    given-names: Frank
+license: AGPL-3.0
+version: _VERSION_
+repository-code: "https://github.com/delb-py/delb"
```
Comment (JKatzwinkel) [inline CITATION.cff.tmpl:12]:
```suggestion
repository-code: "https://github.com/delb-xml/delb-py"
```

File: Justfile [hunk 13]
```diff
@@ -34,16 +36,25 @@ pytest:
 	python -m pytest --cov-config .coveragerc --cov=_delb --cov=delb tests
 
 # release the current version on github & the PyPI
-release: tests
-	git tag -f {{version}}
-	git push origin main
-	git push -f origin {{version}}
-	poetry publish --build
+release: test
```
Comment (JKatzwinkel) [inline Justfile:39]:
```suggestion
release: tests
```

**Review by funkyfuture — APPROVED** [review]

**Pull request merged**